XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Simulation/BaseSimulatedControllerDataProvider.cs

```diff
@@ -0,0 +1,261 @@
+            DepthMultiplier = profile.DepthMultiplier;
+            JitterAmount = profile.JitterAmount;
+            ToggleLeftPersistentKey = profile.ToggleLeftPersistentKey;
+            LeftControllerTrackedKey = profile.LeftControllerTrackedKey;
+            ToggleRightPersistentKey = profile.ToggleRightPersistentKey;
+            RightControllerTrackedKey = profile.RightControllerTrackedKey;
+            RotationSpeed = profile.RotationSpeed;
+        }
+
+        private SimulationTimeStampStopWatch simulatedUpdateStopWatch;
+        private long lastSimulatedUpdateTimeStamp = 0;
+
+        /// <summary>
+        /// Internal read/write list of currently simulated controllers.
+        /// </summary>
+        protected List<BaseController> SimulatedControllers { get; } = new List<BaseController>();
+
+        /// <summary>
+        /// Gets or sets whether the left controller is currently set to be always visible
+        /// no matter the simulated tracking state.
+        /// </summary>
+        protected bool LeftControllerIsAlwaysVisible { get; set; }
+
+        /// <summary>
+        /// Gets or sets whether the right controller is currently set to be always visible
+        /// no matter the simulated tracking state.
+        /// </summary>
+        protected bool RightControllerIsAlwaysVisible { get; set; }
+
+        /// <summary>
+        /// Gets or sets whether the left controller is currently tracked by simulation.
+        /// </summary>
+        protected bool LeftControllerIsTracked { get; set; }
+
+        /// <summary>
+        /// Gets or sets whether the right controller is currently tracked by simulation.
+        /// </summary>
+        protected bool RightControllerIsTracked { get; set; }
+
+        /// <inheritdoc />
+        public double SimulatedUpdateFrequency { get; }
+
+        /// <inheritdoc />
+        public float ControllerHideTimeout { get; }
+
+        /// <inheritdoc />
+        public float DefaultDistance { get; }
+
+        /// <inheritdoc />
+        public float DepthMultiplier { get; }
+
+        /// <inheritdoc />
+        public float JitterAmount { get; }
+
+        /// <inheritdoc />
+        public KeyCode ToggleLeftPersistentKey { get; }
+
+        /// <inheritdoc />
+        public KeyCode LeftControllerTrackedKey { get; }
+
+        /// <inheritdoc />
+        public KeyCode ToggleRightPersistentKey { get; }
+
+        /// <inheritdoc />
+        public KeyCode RightControllerTrackedKey { get; }
+
+        /// <inheritdoc />
+        public float RotationSpeed { get; }
+
+        /// <summary>
+        /// Gets a read only list of active controllers. This property hides the inherited
+        /// active controllers property.
+        /// </summary>
+        /// <remarks>Subject to change, once the new controller refactorings are in place.</remarks>
+        public new IReadOnlyList<BaseController> ActiveControllers => SimulatedControllers;
```

Comment:
why are we using the new keyword to override this?

Reply:
This is essentially what we talked about the other day. The new keyword is required to override the base controller data provider property which is not used by this data provider (or any controller data provider most likely). I need the property to point to my own controller registriy in the derived type.

Reply:
We can revisit this later.